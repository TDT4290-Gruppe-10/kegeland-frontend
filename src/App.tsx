import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { Box } from '@chakra-ui/react';
import { useRoutes } from 'react-router-dom';

import { store, persistor } from './state/store';
import withSilentRefresh from './hoc/withSilentRefresh';
import routes from './routes';

const Wrapper = withSilentRefresh(Box, {
  backgroundColor: 'gray.100',
  height: 'full',
  width: 'full',
});

function App() {
  const renderRoutes = useRoutes(routes);
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Wrapper>{renderRoutes}</Wrapper>
      </PersistGate>
    </Provider>
  );
}

export default App;
