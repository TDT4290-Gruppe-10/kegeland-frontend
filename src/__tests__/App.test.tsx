import App from '../App';
import renderer from 'react-test-renderer';
import theme from '../constants/theme.constants';
import { MemoryRouter } from 'react-router-dom';
import { ChakraProvider } from '@chakra-ui/react';

describe('App', () => {
  it('should renders correctly', () => {
    const page = renderer.create(
      <MemoryRouter>
        <ChakraProvider theme={theme}>
          <App />
        </ChakraProvider>
      </MemoryRouter>,
    );
    expect(page.toJSON()).toMatchSnapshot();
  });
});
