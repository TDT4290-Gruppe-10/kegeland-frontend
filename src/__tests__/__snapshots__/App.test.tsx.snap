// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should renders correctly 1`] = `<span />`;
